fastapi_jsonrpc/__init__.py

```diff
@@ -1318,6 +1321,77 @@ def openapi(self):
                     result['paths'][route.path][media_type]['responses'].pop('default', None)
         return result
 
+    def openrpc(self):
+        methods_spec = []
+        schemas_spec = {}
+        ref_template = '#/components/schemas/{model}'
+
+        for route in self.routes:
+            if not isinstance(route, MethodRoute):
+                continue
+
+            params_schema = route.params_model.schema(ref_template=ref_template)
+
+            if isinstance(route.result_model, BaseModel):
+                result_schema = route.result_model.schema(ref_template=ref_template)
+            else:
+                result_model = create_model(f'{route.name}_Result', result=(route.result_model or Any, ...))
+                result_schema = result_model.schema(ref_template=ref_template)
+
+            method_spec = {
+                'name': route.name,
+                'summary': route.summary,
+                'params': [
+                    {
+                        'name': param_name,
+                        'schema': param_schema,
+                        'required': param_name in params_schema.get('required', []),
+                    }
+                    for param_name, param_schema in params_schema['properties'].items()
+                ],
+                'result': {
+                    'name': result_schema['title'],
+                    'schema': result_schema['properties']['result'],
+                },
+                'tags': [
+                    {
+                        'name': tag,
+                    }
+                    for tag in route.tags
+                ],
+                'errors': [
+                    {
+                        'code': error.CODE,
+                        'message': error.MESSAGE,
+                    }
+                    for error in route.errors
```

Comment:
What about more complicated error schema? OpenRPC allow use extra fields except `code` and `message`?

Right now fastapi-jsonrpc support custom ErrorModel which allow you extend jsonrpc error results and this custom schema is visible in openapi (Swagger)

Reply:
Yep, https://spec.open-rpc.org/#error-object allow custom data too

`BaseError` have a get_data_model() method. You can use it to pass correct schema into openRPC data part

Reply:
Thanks, custom errors' schemas are now supported https://github.com/smagafurov/fastapi-jsonrpc/pull/56/commits/818c1582fa5123f50298fca0157570027b96ed00